### DISCUSSION TITLE
Questions for challenge 2

### DISCUSSION THREAD

**Discussion by micsc12** [surgical-robotics-ai/surgical_robotics_challenge#36] [Q&A] [answered]
1. It says that Ground truth pose of needle will be available. Will this information be realtime, or only the starting pose? 
2. Do you have any updates on the magnitude of the PSM inaccuracy during evaluation? 
3. Finally, I'm a bit confused about the 5 mm of the needle tip that must be visible beyond the exit point. Is this 5 mm in the simulation (where everything is scaled to be a lot larger than live) or 5 mm in real-life scale? Also, do you know at what the scale the simulation is wrt. real world?

**Comment by adnanmunawar** [comment, marked as answer]
Hello, 
These are the answers to your questions:
> It says that Ground truth pose of needle will be available. Will this information be realtime, or only the starting pose?

We shall provide only the starting pose and not the pose of the needle in real-time.

>Do you have any updates on the magnitude of the PSM inaccuracy during evaluation?

Yes, the first two joints of the PSM will have a max error of 5 degrees, and the third joint will have a max error of 5mm.

>Finally, I'm a bit confused about the 5 mm of the needle tip that must be visible beyond the exit point. Is this 5 mm in the simulation (where everything is scaled to be a lot larger than live) or 5 mm in real-life scale? Also, do you know at what the scale the simulation is wrt. real world?

Good point, as you noticed, the simulation is scaled up. For reference 1 mm, (or 0.001 m) is 0.01 Simulation Units. Thus 5 mm error would correspond to 0.05 in the simulation. We have updated the [Challenge Task 2 description](https://collaborative-robotics.github.io/surgical-robotics-challenge/challenge-2.html) and added a figure to make this clear.

**Comment by micsc12** [comment]
Hi @adnanmunawar 
I have another question for you :) 

For the evaluation, ground truth data will provided for the needle pose, and sutur entry and exit points. 
I expect the needle pose is gonna be published on topic /CRTK/Needle/measured_cp, but 4 different topics exists for Entry and Exit points, and i'm not sure which one to use. 

Can we just use /CRTK/Entry1/measured_cp ?

//Michael

**Reply by adnanmunawar** [reply]
Hello Michael,
For task 2, you may pick any pair of Entry and Exit points to perform the drive-through task into the tissue. We can figure out your selected entry/exits based on the needle trajectory and its contacts.

**Reply by micsc12** [reply]
Thanks :)
